### PR TITLE
Add attributes in VOC format

### LINKED ISSUE #1518: Pascal VOC 2012 dataset export does not export Attributes correctly
### Expected Behaviour
I am labelling a "person" with an attribute "pose" which can be either "y", "n" or "o". I expect this attribute to be accessible in the produced annotation file. I export as a dataset of pascal voc 2012 format and I am in labelling a video.
Screenshot from 2020-05-09 11-04-18

### Current Behaviour
The following annotation is being produced where the "pose" field is set to "Unspecified" instead of "o".
```xml
<annotation>
	<folder/>
	<filename>52.jpg</filename>
	<source>
		<database>Unknown</database>
		<annotation>Unknown</annotation>
		<image>Unknown</image>
	</source>
	<size>
		<width>1920</width>
		<height>1080</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>person</name>
		<pose>Unspecified</pose>
		<occluded>0</occluded>
		<bndbox>
			<xmin>970.3649338942307</xmin>
			<ymin>70.13979867788461</ymin>
			<xmax>1003.2655636714069</xmax>
			<ymax>112.42590015117933</ymax>
		</bndbox>
	</object>
</annotation>
```

### Possible Solution
Append the attribute to the label to produce three labels person_o, person_y, person_n. Not nice and requires postprocessing after the labelling.

### Your Environment

- Git hash commit (`git log -1`): 034268e87dbf72b6ca08f017152b962882cc1d84
- Docker version `docker version`: Docker 19.03.8
- Are you using Docker Swarm or Kubernetes? No
- Operating System and version (e.g. Linux, Windows, MacOS): Ubuntu 18.04

### PR DESCRIPTION
**Pull request by zhiltsov-max**
### Motivation and context

Closes #1518 

Allows any values in `pose` object attribute.
Allows to read and write annotation attributes in `attributes` element of VOC XML annotations.

Exporting is enabled by default and can be disabled with `--allow-attributes=False` CLI option of Datumaro export.

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [x] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by nmanovic** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- datumaro/datumaro/plugins/voc_format/extractor.py  2
- datumaro/datumaro/plugins/voc_format/converter.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5741738&bid=18783357)

**Comment by azhavoro** [comment]
@zhiltsov-max 1 issue is here:
the value of checkbox attribute is dumped as `True/False` and imported as is, so that cannot be handled by UI properly. However for the CVAT format, the lower case is used: https://github.com/opencv/cvat/blob/develop/datumaro/datumaro/plugins/cvat_format/converter.py#L284

**Comment by nmanovic** [comment]
@azhavoro , could you please look?

File: cvat/apps/dataset_manager/bindings.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 import datumaro.components.extractor as datumaro
 from cvat.apps.engine.frame_provider import FrameProvider
-from cvat.apps.engine.models import AttributeType, ShapeType
+from cvat.apps.engine.models import AttributeType, ShapeType, AttributeSpec
```
Comment (nmanovic) [inline cvat/apps/dataset_manager/bindings.py:14]: Codacy found an issue: [Unused AttributeSpec imported from cvat.apps.engine.models](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5741738)

**Comment by nmanovic** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- datumaro/datumaro/plugins/voc_format/extractor.py  2
- datumaro/datumaro/plugins/voc_format/converter.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5741738&bid=18783357)

**Comment by zhiltsov-max** [comment]
I don't know if we should fail or just log if there is an incorrect attribute value.

**Comment by nmanovic** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- datumaro/datumaro/plugins/voc_format/extractor.py  2
- datumaro/datumaro/plugins/voc_format/converter.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5741738&bid=18783357)

**Comment by azhavoro** [comment]
@zhiltsov-max could you fix tests?

**Comment by nmanovic** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- datumaro/datumaro/plugins/voc_format/converter.py  4
- datumaro/datumaro/plugins/voc_format/extractor.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5741738&bid=18783357)

**Comment by zhiltsov-max** [comment]
@azhavoro, done.

**Pull request merged**